react.tsx

```diff
@@ -135,37 +135,37 @@ export function useFlags<F extends string=string, T extends string=string>(_flag
     const traits = useConstant<string[]>(flagsAsArray(_traits))
     const flagsmith = useContext(FlagsmithContext)
     const [renderRef, setRenderRef] = useState(getRenderKey(flagsmith as IFlagsmith, flags, traits));
-
     const eventListener = useCallback(() => {
         const newRenderKey = getRenderKey(flagsmith as IFlagsmith, flags, traits)
         if (newRenderKey !== renderRef) {
             flagsmith?.log("React - useFlags flags and traits have changed")
             setRenderRef(newRenderKey)
         }
-    },[renderRef])
+    }, [renderRef])
+    const emitterRef = useRef(events.once('event', eventListener));
+
+
 
-    events.once('event', eventListener)
-    
     if (firstRender.current) {
         firstRender.current = false;
         flagsmith?.log("React - Initialising event listeners")
     }
 
     useEffect(()=>{
         return () => {
-            events.off('event', eventListener)
```

Comment:
This was incorrectly removing multiple listeners, new approach only removes the ref's listener